src/makeRobot.js

```diff
@@ -38,7 +38,81 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  return {
+    name: name,
+    wheels: wheels,
+    version: version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, chip version: `
+      + `${this.version}, wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goForward(number) {
+      if (number > 0) {
+        this.coords.y += number;
+
+        return this;
+      } else if (!number) {
+        this.coords.y++;
+
+        return this;
+      } else {
+        return this;
+      }
+    },
```

Comment:
lets avoid such complicated if-else structures. One-level if is enough for most cases
here you can set default value for parameter and work with it
```suggestion
    goForward(step = 1) {
     if (step > 0) {
        this.coords.y += step;
      }

      return this;
    },
```